src/field.rs

```diff
@@ -279,7 +290,12 @@ macro_rules! make_field {
                 // The fields included in this comparison MUST match the fields
                 // used in Hash::hash
                 // https://doc.rust-lang.org/std/hash/trait.Hash.html#hash-and-eq
-                $fp.residue(self.0) == $fp.residue(rhs.0)
+
+                // Check the invariant that the integer representation is fully reduced.
+                assert!(self.0 < $fp.p);
+                assert!(rhs.0 < $fp.p);
```

Comment:
Is it feasible/safe to only run these checks in tests? Would be nice to safe a few cycles here. (Similar question below.)

Reply:
Sure, I can swap it out for `debug_assert!`